src/main.ts

```diff
@@ -178,7 +178,8 @@ export class Minifier {
       let childEnd = child.getEnd() - node.getStart();
       output += nodeText.substring(prevEnd, childStart);
       let childText = '';
-      if (renameIdent && child.kind === ts.SyntaxKind.Identifier) {
+      let hasName = (<any>node).name;
```

Comment:
move outside the forEach loop (it only depends on the node, not child), also rename `hasName` is a name of a boolean, but this more like `nameChildNode`.